### PR TITLE
add docker-compose configuration; update config loading code

### PR DESCRIPTION
**Pull request by DerekStrickland**
## Summary

This PR introduces a `docker-compose` feature to bootstrap a local, observable development environment.

### Added

- Introduces a `docker-compose` configuration in the `devenv` folder for running an observable harbor stack for local development.
- Adds a `db_connection` method to make use of the `DB_CONNECTION` environment variable. Defaults to the `docker-compose` instance if not set.

### Changed

- Updates the README to include developer machine dependencies and an environment variables section.

## How to test

- Check out the branch
- cd to `devenv` and run `docker compose up`
- cd back to the root directory and run `cargo test`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,3 +34,5 @@ cargo doc --no-deps
 
 Documentation for each crate will be generated and output to the `target/doc` subdirectory.
 
+
+
```
Comment (sbolel) [inline README.md:38]: nitpick: trim empty eof lines

```suggestion
```

File: platform/tests/common/mongodb.rs [hunk 2]
```diff
@@ -10,38 +8,14 @@ use platform::mongodb::{Context, MongoDocument, Store};
...
-    pub fn connection_string() -> String {
-        "mongodb://localhost:27017".to_string()
-    }
-}
-
 #[allow(dead_code)]
 pub async fn local_context() -> Result<Context, Error> {
     let ctx = Context{
-        connection_uri: LocalContext::connection_string(),
+        connection_uri: "mongodb://localhost:27017".to_string(),
```
Comment (sbolel) [inline platform/tests/common/mongodb.rs:18]: is it possible to avoid hardcoding this?

**Review by sbolel — COMMENTED** [review]
left minor comments!

File: platform/tests/common/mongodb.rs (hunk 2 above)
Comment (DerekStrickland) [reply platform/tests/common/mongodb.rs:18]: Yeah, I can do something similar to core.

FYI I am expecting a bit of this to change when we start integrating with CI.

Comment (sbolel) [reply platform/tests/common/mongodb.rs]: Sweet!

File: README.md [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+- [Protobuf](https://grpc.io/docs/protoc-installation/)
+- [docker](https://docs.docker.com/get-docker/) (optional - used for tests)
+- [docker compose](https://docs.docker.com/compose/install/) (optional - used for local environment)
+
+## Environment
+
+The following environment variables are referenced in code. When possible, defaults are provided that
+support the `docker-compose` configuration found in the `devenv` folder.
+
+- `DB_CONNECTION` - Mongo connection string. Defaults to `mongodb://localhost:27017`.
```
Comment (DerekStrickland) [inline README.md:22]:
```suggestion
- `DB_CONNECTION` - Mongo connection string.
```

**Review by sbolel — CHANGES_REQUESTED** [review]
Running `cargo test` fails with the following errors:

```
     Running tests/mongodb.rs (target/debug/deps/mongodb-e9df59552daa571c)

running 10 tests
test can_assert_deny_for_disabled_user ... ok
test can_assert_deny_for_readonly_policy ... ok
test can_assert_allow_any_for_admin ... ok
test can_assert_deny_any_for_disabled_admin ... ok
test can_crud ... ok
test can_assert_implicit_deny_no_action ... FAILED
test can_assert_implicit_deny_no_policy ... FAILED
test can_assert_explicit_deny ... FAILED
test can_create_local_cluster ... FAILED
test can_assert_explicit_allow_for_action ... FAILED

failures:

---- can_assert_implicit_deny_no_action stdout ----
thread 'can_assert_implicit_deny_no_action' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', platform/tests/common/mongodb.rs:18:56

---- can_assert_implicit_deny_no_policy stdout ----
thread 'can_assert_implicit_deny_no_policy' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', platform/tests/common/mongodb.rs:18:56

---- can_assert_explicit_deny stdout ----
thread 'can_assert_explicit_deny' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', platform/tests/common/mongodb.rs:18:56
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- can_create_local_cluster stdout ----
thread 'can_create_local_cluster' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', platform/tests/common/mongodb.rs:18:56

---- can_assert_explicit_allow_for_action stdout ----
thread 'can_assert_explicit_allow_for_action' panicked at 'called `Result::unwrap()` on an `Err` value: NotPresent', platform/tests/common/mongodb.rs:18:56


failures:
    can_assert_explicit_allow_for_action
    can_assert_explicit_deny
    can_assert_implicit_deny_no_action
    can_assert_implicit_deny_no_policy
    can_create_local_cluster

test result: FAILED. 5 passed; 5 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

error: test failed, to rerun pass `-p platform --test mongodb`
```

**Review by qtpeters — APPROVED** [review]
Looks good!

**Comment by DerekStrickland** [comment]
Closed in favor of #59

**Pull request closed without merging**